### PR TITLE
Add more NMEA 0183 conversions 

### PR DESCRIPTION
**Pull request by sbender9**
Added Heading True, Heading Magnetic, Rate of Turn, Depth Below Keel, Depth Below Surface, Depth Below Transducer and Water Temperature

Adding these so we can fully support the iSailor iOS application

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -67,6 +103,27 @@ module.exports = function(app) {
     if (options.PSILCD1) {
       mapToNmea(PSILCD1)
     }
+        if (options.HDT) {
```
Comment (tkurki) [inline index.js]: indenting

File: index.js [hunk 2]
```diff
@@ -67,6 +103,27 @@ module.exports = function(app) {
...
+    }
+    if (options.DBS) {
+      mapToNmea(DBS)
+    }
+    if (options.DBT) {
+      mapToNmea(DBT)
+    }
+    if (options.MTW) {
+      mapToNmea(MTW)
+    }
```
Comment (tkurki) [inline index.js]: This list is a bit stupid: we could iterate over `Object.keys(plugin.schema.properties)`.

Comment (sbender9) [inline index.js]: Not sure if my change is right. I could not figure out how to reference those module vars like 'APB', so I moved them up on the plugin

Comment (tkurki) [inline index.js]: Lgtm

**Comment by tkurki** [comment]
I would probably use a single global constant `sentences` that holds structures like 
```
{
  description: "Heading True",
  keys: [...],
  f: () => {}
}
```
keyed by NMEA three letter keys and produce the schema from that, since the schema entries are all the same.

This way it is blatantly obvious what you need to do to add one more sentence.

You want to to fiddle with this or should I just merge?

**Comment by sbender9** [comment]
Good idea. I can do it.

File: index.js [hunk 3]
```diff
@@ -52,225 +27,351 @@ module.exports = function(app) {
...
-      '0000',
-      '7.0',
-      'E'
-    ]);
-  }
-}
+  //===========================================================================
+  
+  Object.keys(plugin.sentences).forEach(key => {
+    sentence = plugin.sentences[key]
```
Comment (tkurki) [inline index.js]: This variable is leaking to the global scope.

**Pull request merged**